tests/garage/torch/policies/test_tanh_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,131 @@
+        act_dim = env_spec.action_space.flat_dim
+        obs = torch.ones([batch_size, obs_dim], dtype=torch.float32)
+        init_std = 2.
+
+        policy = TanhGaussianMLPPolicy(env_spec=env_spec,
+                                       hidden_sizes=hidden_sizes,
+                                       init_std=init_std,
+                                       hidden_nonlinearity=None,
+                                       std_parameterization='exp',
+                                       hidden_w_init=nn.init.ones_,
+                                       output_w_init=nn.init.ones_)
+
+        expected_mean = torch.full([batch_size, act_dim], 1.0)
+        action, prob = policy.get_actions(obs)
+        assert np.allclose(prob['mean'], expected_mean.numpy(), rtol=1e-3)
+        assert action.shape == (batch_size, act_dim)
+
+    # yapf: disable
+    @pytest.mark.parametrize('batch_size, hidden_sizes', [
+        (1, (1, )),
+        (6, (3, )),
+        (11, (6, )),
+        (25, (3, 5)),
+        (34, (2, 10, 11)),
+    ])
+    # yapf: enable
+    def test_is_pickleable(self, batch_size, hidden_sizes):
+        """Test if policy is unchanged after pickling."""
+        env_spec = TfEnv(DummyBoxEnv())
+        obs_dim = env_spec.observation_space.flat_dim
+        obs = torch.ones([batch_size, obs_dim], dtype=torch.float32)
+        init_std = 2.
+
+        policy = TanhGaussianMLPPolicy(env_spec=env_spec,
+                                       hidden_sizes=hidden_sizes,
+                                       init_std=init_std,
+                                       hidden_nonlinearity=None,
+                                       std_parameterization='exp',
+                                       hidden_w_init=nn.init.ones_,
+                                       output_w_init=nn.init.ones_)
+
+        output1_action, output1_prob = policy.get_actions(obs)
+
+        p = pickle.dumps(policy)
+        policy_pickled = pickle.loads(p)
+        output2_action, output2_prob = policy_pickled.get_actions(obs)
+        assert np.allclose(output2_prob['mean'],
+                           output1_prob['mean'],
+                           rtol=1e-3)
+        assert output1_action.shape == output2_action.shape
+
+    def test_is_vectorized(self):
+        """Test Tanh Gaussian Policy is vectorized."""
+        env_spec = TfEnv(DummyBoxEnv())
+        init_std = 2.
+
+        policy = TanhGaussianMLPPolicy(env_spec=env_spec,
+                                       hidden_sizes=(1, ),
+                                       init_std=init_std,
+                                       hidden_nonlinearity=None,
+                                       std_parameterization='exp',
+                                       hidden_w_init=nn.init.ones_,
+                                       output_w_init=nn.init.ones_)
+        assert policy.vectorized
+
+    def test_to(self):
+        """Test Tanh Gaussian Policy can be moved to cpu."""
```

Comment:
Is this really the best we can do? Isn't it already on the cpu? Would this test actually catch a bug in `.to()`?

It would probably be better to make a test that starts `if torch.cuda.is_available():`, and then makes sure that `.get_action` still works after moving the policy to `cuda:0`.

Reply:
I initially was unsure of how to go about this because we don't have gpu's on travis. I'll go ahead and implement your idea.